src/test/regress/expected/isolation_index_vs_all_on_mx.out

```diff
@@ -0,0 +1,333 @@
+
+run_commands_on_session_level_connection_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+step w2-create-index-concurrently:
+ SELECT run_commands_on_session_level_connection_to_node('CREATE INDEX CONCURRENTLY dist_table_index_2 ON dist_table(id)');
+
+run_commands_on_session_level_connection_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+step w1-stop-connection:
+ SELECT stop_session_level_connection_to_node();
+
+stop_session_level_connection_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+step w2-stop-connection:
+ SELECT stop_session_level_connection_to_node();
+
+stop_session_level_connection_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+step coord-print-index-count:
+ SELECT
+  result
+ FROM
+  run_command_on_placements('dist_table', 'select count(*) from pg_indexes WHERE tablename = ''%s''')
+ ORDER BY
+  nodeport;
+
+result
+---------------------------------------------------------------------
+     2
+     2
+     2
+     2
+(4 rows)
+
+
+starting permutation: w1-start-session-level-connection w1-create-named-index w1-begin-on-worker w1-delete coord-begin coord-short-statement-timeout coord-take-lock w1-reindex deadlock-checker-call coord-rollback w1-commit-worker w1-stop-connection
+step w1-start-session-level-connection:
+ SELECT start_session_level_connection_to_node('localhost', 57637);
+
+start_session_level_connection_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+step w1-create-named-index:
+ SELECT run_commands_on_session_level_connection_to_node('CREATE INDEX dist_table_index ON dist_table(id)');
+
+run_commands_on_session_level_connection_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+step w1-begin-on-worker:
+ SELECT run_commands_on_session_level_connection_to_node('BEGIN');
+
+run_commands_on_session_level_connection_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+step w1-delete:
+ SELECT run_commands_on_session_level_connection_to_node('DELETE FROM dist_table');
+
+run_commands_on_session_level_connection_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+step coord-begin:
+ BEGIN;
+
+step coord-short-statement-timeout:
+ SET statement_timeout = 100;
+
+step coord-take-lock:
+ LOCK TABLE dist_table IN ACCESS EXCLUSIVE MODE;
+ <waiting ...>
+step w1-reindex: 
+ SELECT run_commands_on_session_level_connection_to_node('REINDEX INDEX dist_table_index');
+
+run_commands_on_session_level_connection_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+step deadlock-checker-call:
+  SELECT check_distributed_deadlocks();
+
+check_distributed_deadlocks
+---------------------------------------------------------------------
+f
```

Comment:
I would be happier if distributed deadlock detection actually worked in this scenario. Instead coordinator waits indefinitely and gets cancelled after statement_timeout.

The problem here is that we disable distributed deadlock detection in our regression suite. However it is expected observe the deadlock after we call this UDF explicitly. I am fairly certain that this deadlock is detected by our daemon, but I failed to have this test case behave properly.